addons/interact_menu/functions/fnc_addActionToClass.sqf

```diff
@@ -25,22 +26,27 @@ if (!params [["_objectType", "", [""]], ["_typeNum", 0, [0]], ["_parentPath", []
     ERROR("Bad Params");
     []
 };
-TRACE_4("addActionToClass",_objectType,_typeNum,_parentPath,_action);
+private _useInheritance = _this param [4, false, [false]];
+private _excludedClasses = _this param [5, [], [[]]];
+TRACE_6("addActionToClass",_objectType,_typeNum,_parentPath,_action,_useInheritance,_excludedClasses);
 
-if (param [4, false, [false]]) exitwith {
+if (_useInheritance) exitwith {
     BEGIN_COUNTER(addAction);
+    private _cfgVehicles = configFile >> "CfgVehicles"; // store this so we don't resolve for every element
+    _excludedClasses = _excludedClasses apply {configName (_cfgVehicles >> _x)}; // ends up being faster than toLower'ing everything else
     if (_objectType == "CAManBase") then {
-        GVAR(inheritedActionsMan) pushBack [_typeNum, _parentPath, _action];
+        GVAR(inheritedActionsMan) pushBack [_typeNum, _parentPath, _action, _excludedClasses];
         {
```

Comment:
Don't we need a check, like you did in `preInit`?
`_excludedClasses findIf {_type isKindOf _x} == -1`

Reply:
We indeed need to add it:
Let's say `C_man_1` is already initialised (I was in the ACE arsenal). You want to add an action to `CAManBase`, but want to exclude `Civilian_F` and its children. `C_man_1` inherits from `Civilian_F`.
If you use the existing code, it adds it to `(GVAR(inheritedClassesMan) - _excludedClasses)`, which would still contain `C_man_1`, as `_excludedClasses` contains only `Civilian_F`, meaning the action would be added to `C_man_1`.